### DISCUSSION TITLE
.subquery() cancels the load_only() option

### DISCUSSION THREAD

**Discussion by bfontaine** [sqlalchemy/sqlalchemy#11937] [Usage Questions]
Hello,
I’m using SQLAlchemy 2.0.35 with Python 3.12.6.

I have a query with a subquery and I’m trying to reduce the fields it fetches, but for some reason `load_only` seems to be canceled by the `.subquery()` call:

```python
class Item(Base):
    __tablename__ = "items"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    code: Mapped[str] = mapped_column(String, unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String, nullable=False)
    weight: Mapped[float] = mapped_column(Double, nullable=False)
```

`load_only` works on a simple select:
```python
stmt = select(Item).options(load_only(Item.code))
print(stmt)
# => SELECT items.id, items.code 
#    FROM items
```

But if I subquery/alias it it has no effect:
```python
a = aliased(Item, stmt.subquery())
stmt = select(a)
print(stmt)

# => SELECT anon_1.id, anon_1.code, anon_1.name, anon_1.weight 
#    FROM (SELECT items.id AS id, items.code AS code, items.name AS name, items.weight AS weight 
#    FROM items) AS anon_1
```

If I `load_only` on the outer query it works (but does not affect the subquery, as expected):
```python
print(stmt.options(load_only(a.code)))
# => SELECT anon_1.id, anon_1.code 
#    FROM (SELECT items.id AS id, items.code AS code, items.name AS name, items.weight AS weight 
#    FROM items) AS anon_1
```

(this is the same issue as [this 2022 StackOverflow question](https://stackoverflow.com/q/74384164/735926))

What am I missing? In the last query it does not make sense to select `name` and `weight` in the subquery because they are discarded in the outer query. How can I make this work?
Thanks in advance

**Comment by bfontaine** [comment]
I tried with `with_only_columns(…)` instead of `.options(load_only(…))` in the subquery, but then that breaks the joins I make in the outer query (not shown in my small example) because for some reason the `ON parents.id = anon_1.parent_id` becomes `ON parents.id = items.parent_id`, which is invalid here (`missing FROM-clause entry for table`).

**Comment by CaselIT** [comment]
Hi,

That's expected, the loader option only applies to the outer query, since those are treated as indication to the orm about what to do with the data.

In general the sql engine should be able to automatically ignore those subquery columns. If you want to specify the column the most straightforward way should be `select(Item.id, Item.code).subquery()`.
